src/layout.sh

```diff
@@ -172,23 +172,43 @@ start_listener() {
   set_desktop_option $selected_desktop 'pid'    "$LAYOUT_PID";
 
   # Setup
-  initialize_layout;
+  __initialize_layout;
 
   # Recalculate styles as soon as they are set if it is on the selected desktop
   if [[ "$(get_focused_desktop)" == "$selected_desktop" ]]; then
     # Calculate layout twice to ensure rotations are corrected from previous layout
-    recalculate_layout;
-    recalculate_layout;
+    __recalculate_layout;
+    __recalculate_layout;
   fi;
 
   echo "[$LAYOUT_PID]";
 }
 
 once_layout() {
-  if (echo -e "$BSP_DEFAULT_LAYOUTS" | grep "^$@$"); then exit 0; fi
-  setup_layout "$@";
-  run_layout "$@";
-  run_layout "$@";
+  if (echo -e "$BSP_DEFAULT_LAYOUTS" | grep "^$1$"); then exit 0; fi
+  local focused_desktop=$(get_focused_desktop);
+  local selected_desktop="${2:-$focused_desktop}";
+
+  __calculate_layout() {
+    setup_layout "$@";
+    run_layout "$@";
+    run_layout "$@";
+  }
+
+  if [[ "$selected_desktop" != "$focused_desktop" ]]; then
+    bspc subscribe desktop_focus | while read line; do
+      event=$(echo "$line" | awk '{print $1}');
+      desktop_id=$(echo "$line" | awk '{print $3}');
+      desktop_name=$(get_desktop_name_from_id "$desktop_id");
+
+      if [[ "$desktop_name" = "$selected_desktop" ]]; then
+        __calculate_layout "$@";
+        exit 0
+      fi;
+    done & disown;
+  else
+    __calculate_layout "$@";
+  fi;
```

Comment:
to be honest, i'm not comfortable enough with the code to comment a lot on it...

i do not see any obvious issue with that hunk and it solves 3 issues at the time :open_mouth: 

that is a go for me :ok_hand: